llvm/lib/Target/AMDGPU/AMDGPULowerModuleLDSPass.cpp

```diff
@@ -648,7 +649,19 @@ class AMDGPULowerModuleLDS {
         } else if (K.second.size() == 1) {
           KernelAccessVariables.insert(GV);
         } else if (set_is_subset(K.second, HybridModuleRootKernels)) {
-          ModuleScopeVariables.insert(GV);
+          uint64_t LocalMemLimit = 0;
+          for (Function &F : M) {
+            if (!F.isDeclaration()) {
+              const GCNSubtarget &ST = TM.getSubtarget<GCNSubtarget>(F);
+              LocalMemLimit = ST.getAddressableLocalMemorySize();
```

Comment:
The limit should really come from the entry point kernel, not just the first function you happen to find